src/pages/index.js

```diff
@@ -111,13 +145,68 @@ const addCardModalPopup = new PopupWithForm(".modal_card-add", (data) => {
     link: data.url,
   };
 
-  cardsListSection.addItem(createCard(newCard));
+  addCardModalPopup.setSaveButtonText("Updating");
+
+  api
+    .postCard("/cards", newCard)
+    .then((res) => {
+      console.log(res);
+      const cardElement = createCard(res);
+      cardsListSection.addItem(cardElement);
+    })
+    .then(addCardModalPopup.close())
+    .catch((err) => console.error(err))
+    .finally(addCardModalPopup.setSaveButtonText("Save"));
 });
 
 function createCard(newCard) {
-  const card = new Card(newCard, "#cardElementTemplate", () => {
-    cardOpenModalPopup.open(newCard);
-  });
+  const card = new Card(
+    newCard,
+    "#cardElementTemplate",
+    () => {
+      cardOpenModalPopup.open(newCard);
+    },
+
+    () => {
+      const cardDeleteConfirm = new PopupWithConfirm(
+        ".modal_confirm-delete-card",
+        () => {
+          api
```

Comment:
Needs correction: you should also notify the user about the loading process by changing the button text

Reply:
Done